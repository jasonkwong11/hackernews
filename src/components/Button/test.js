import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import {Button, Loading, ButtonWithLoading} from '../Button';

describe('Button', () => {
  const props = {
    onClick: undefined,
    className: ''
  }

  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Button {...props}>
        Give Me More
      </Button>, div);
  })

  test('snapshots', () => {
    const component = renderer.create(
      <Button {...props}>
        Give Me More
      </Button>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})

describe('Loading', () => {
  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Loading />, div)
  });

  test('snapshots', () => {
    const component = renderer.create(
      <Loading />
    )
    let tree = component.toJSON
    expect(tree).toMatchSnapshot();
  })
})

