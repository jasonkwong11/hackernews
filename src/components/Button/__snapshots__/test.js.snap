// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshots 1`] = `
<button
  className=""
  onClick={undefined}
  type="button"
>
  Give Me More
</button>
`;

exports[`Loading snapshots 1`] = `[Function]`;
